### DISCUSSION TITLE
create/update artifact via api when keycloak is enabled

### DISCUSSION THREAD

**Discussion by meenajaeiei** [Apicurio/apicurio-registry#2911] [Q&A] [answered]
Hello,
I would like to create a software pipeline to store the artifact after the test phase is passed. Are there any ways to create/update artifacts via curl when Keycloak authentication is enabled?

Thanks,

**Comment by carlesarnal** [comment, marked as answer]
Hi @meenajaeiei,

You have two options, to acquire the token on your side and pass it as Authorization header or you can also give the maven plugin a try.

Thanks

**Reply by EricWittmann** [reply]
Link to documentation about the maven plugin (which is what I would recommend):

https://www.apicur.io/registry/docs/apicurio-registry/2.3.x/getting-started/assembly-managing-registry-artifacts-maven.html

If you'd rather use `curl` then as Carles said you have to acquire the token from Keycloak (I don't have a link for this handy) and then pass it via an `Authorization` header.  Here's an example that does the latter:

https://www.apicur.io/registry/docs/apicurio-registry/2.3.x/getting-started/assembly-managing-registry-artifacts-api.html#managing-artifacts-using-rest-api_registry
